hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/client/clustering/plan/strategy/FlinkSizeBasedClusteringPlanStrategy.java

```diff
@@ -70,9 +70,11 @@ protected Stream<HoodieClusteringGroup> buildClusteringGroupsForPartition(String
       // check if max size is reached and create new group, if needed.
       // in now, every clustering group out put is 1 file group.
       if (totalSizeSoFar >= writeConfig.getClusteringTargetFileMaxBytes() && !currentGroup.isEmpty()) {
-        LOG.info("Adding one clustering group " + totalSizeSoFar + " max bytes: "
-            + writeConfig.getClusteringMaxBytesInGroup() + " num input slices: " + currentGroup.size());
-        fileSliceGroups.add(Pair.of(currentGroup, 1));
+        if (currentGroup.size() > 1 || writeConfig.isClusteringSortEnabled()) {
+          LOG.info("Adding one clustering group " + totalSizeSoFar + " max bytes: "
+                  + writeConfig.getClusteringMaxBytesInGroup() + " num input slices: " + currentGroup.size());
+          fileSliceGroups.add(Pair.of(currentGroup, 1));
```

Comment:
At present every clustering group out put is 1 file group, therefore the right of `fileSliceGroups` is always 1 which could set `numOutputFileGroups` to 1 and no need use `Pair`.

Reply:
@zhuanshenbsj1, what do you think about above comment?

Reply:
Like target size is 1G，4 file group input,  these sizes are 256M, 256M, 256M, 512M， out put file group number should be 2(file1: 1G, file2: 256M).   I will modify this function according to spark case.

int numOutputGroups = getNumberOfOutputFileGroups(totalSizeSoFar, writeConfig.getClusteringTargetFileMaxBytes());